### DISCUSSION TITLE
How to bundle openvino with pyinstaller on MacOS arm64?

### DISCUSSION THREAD

**Discussion by cansik** [openvinotoolkit/openvino#25690] [Q&A]
It seems that the pre-built `libopenvino.2400.dylib` file (and other openvino libs) from the pypi wheel is not suitable for bundling on MacOS arm64 platforms due to r_path manipulation issues with `install_name_tool`. This is also commented in the [setup.py](https://github.com/openvinotoolkit/openvino/blob/bb7f0e794539ec15c45f323daddb6d2f195055c6/src/bindings/python/wheel/setup.py#L369):

```python
# set rpath if applicable
if sys.platform != "win32" and comp_data.get("rpath"):
    # after tbb libraries on mac arm64 are signed, setting rpath for them will report error:
    # LC_SEGMENT_64 command 3 fileoff field plus filesize field extends past the end of the file
    if comp == "tbb_libs" and ARCH == "arm64" and sys.platform == "darwin":
        continue

...
```

The problem is that this prevents bundling MacOS applications with pyinstaller that use openvino as a dependency.
What is the recommended way to use pyinstaller with openvino on MacOS arm64?

I am not a 100% sure, but I have the feeling that this issue started with the 2024 release.

**PyInstaller Error Message**

```python
47712 INFO: Building PKG (CArchive) my-software.pkg
Traceback (most recent call last):
  File "/user/venv/lib/python3.11/site-packages/PyInstaller/building/utils.py", line 314, in process_collected_binary
    osxutils.set_dylib_dependency_paths(cached_name, target_rpath)
  File "/user/venv/lib/python3.11/site-packages/PyInstaller/utils/osx.py", line 423, in set_dylib_dependency_paths
    _set_dylib_dependency_paths(filename, target_rpath)
  File "/user/venv/lib/python3.11/site-packages/PyInstaller/utils/osx.py", line 543, in _set_dylib_dependency_paths
    raise SystemError(
SystemError: install_name_tool command (['install_name_tool', '-delete_rpath', '@loader_path', '-add_rpath', '@loader_path/../..', '/tmp/arm64/adhoc/no-entitlements/openvino/libs/libopenvino.2400.dylib']) failed with error code 1!
output: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/install_name_tool: object: /tmp/arm64/adhoc/no-entitlements/openvino/libs/libopenvino.2400.dylib truncated or malformed object (/user)
```

**Comment by bibekyess** [comment]
Hi @cansik! 
Did you find any workaround for this?

**Reply by cansik** [reply]
No, not yet. We may have to elevate this to a bug as it only happens on MacOS Arm64 and openvino.

**Reply by bibekyess** [reply]
Right, I also tested in windows and ubuntu. There is no issue on bundling openvino in these platforms.

**Reply by ilya-lavrenov** [reply]
Do you have steps how to reproduce the issue? I can try it locally

**Reply by bibekyess** [reply]
@ilya-lavrenov Please find it below. Thanks! :)

**Comment by bibekyess** [comment]
## Bug- Minimal Reproduction Steps
### **Files:**

1. **`main.py`**:
    ```python
    import openvino

    print("Openvino is successfully imported!!")
    ```

2. **`build_executable.sh`**:
    ```bash
    #!/bin/bash

    # Create and activate a Python virtual environment
    python3 -m venv env
    source env/bin/activate

    pip install openvino pyinstaller
    pyinstaller main.py
    ```

### **Steps to Reproduce:**

   - Save the provided `main.py` and `build_executable.sh` script in your working directory.
   - Execute the build script
     ```bash
     chmod +x build_executable.sh
     ./build_executable.sh
     ```

**Comment by ilya-lavrenov** [comment]
Since OpenVINO packages are signed, looks like you cannot use them with PyInstaller.
But you can build your own OpenVINO package and use it because it's not signed.

**Reply by cansik** [reply]
Building openvino from source to be able to ship the binaries with pyinstaller is quite an overhead (complexity and CI hours). Are you sure that is the issue?

The problem can be reduced to the following command. The other dylibs of openvino are not affected, as far as I can see it's this specific dylib `libopenvino_tensorflow_frontend.2460.dylib`.

```bash
install_name_tool -delete_rpath @loader_path -add_rpath @loader_path/../.. openvino/libs/libopenvino_tensorflow_frontend.2460.dylib
```

This leads to the following error message:

> /usr/bin/install_name_tool: object: openvino/libs/libopenvino_tensorflow_frontend.2460.dylib truncated or malformed object (LC_SEGMENT_64 command 3 fileoff field plus filesize field extends past the end of the file)

**Reply by ilya-lavrenov** [reply]
maybe you know how to remove signature before `install_name_tool` is used?
